### PR TITLE
nvme: correct comments from PR #38

### PR DESCRIPTION
**Pull request by jmpesp**
The fix from #38 was good but I interpreted the spec incorrectly. Correct the comments to explain how the fix actually works.

### CODE REVIEW

**Review by luqmana — APPROVED** [review]

File: propolis/src/hw/nvme/cmds.rs [hunk 1]
```diff
@@ -607,24 +607,17 @@ impl PrpIter<'_> {
...
-                    /*
-                     * The first PRP List entry (i.e. the first pointer to a memory page containing
-                     * additional PRP entries) that if present is contained in the PRP Entry 2 location
-                     * within the command, shall be Qword aligned and may also have a non-zero offset within
-                     * the memory page.
-                     */
+                    // The first PRP List entry (i.e. the first pointer to a memory page containing
+                    // additional PRP entries) that if present is contained in the PRP Entry 2
+                    // location within the command, shall be Qword aligned and may also have a
+                    // non-zero offset within the memory page.
```
Comment (pfmooney) [inline propolis/src/hw/nvme/cmds.rs:613]: The phrasing on this is a bit awkward.  It seems like it needs at least a comma, and perhaps be split up.

File: propolis/src/hw/nvme/cmds.rs [hunk 2]
```diff
@@ -607,24 +607,17 @@ impl PrpIter<'_> {
...
-                     * If this entry is
-                     *
-                     * - not the first PRP entry in the command, or
-                     * - in the PRP List
-                     *
-                     * then the Offset portion of this field shall be cleared to 0h.
-                     */
+                    // PRP2 is allowed to have a non-zero offset into the memory page, but mask it
+                    // out when creating the base part of the PrpNext::List enum - the offset will
+                    // be passed with idx.
```
Comment (pfmooney) [inline propolis/src/hw/nvme/cmds.rs:620]: The phrasing is confusing here too.  "Allowed to have a non-zero offset, but mask it out" seems almost like it's being discarded.  Perhaps make it clear that the list can start at an offset in the page, but must not cross the page boundary, and that any offset is preserved to index into that region.

**Review by pfmooney — APPROVED** [review]
Thanks for making those clearer.

One minor style request: Make the commit message more explanatory about the change, rather than referencing the previous PR - something like "nvme: clarify PrpIter behavior for PRP Lists " or whatever

**Comment by jmpesp** [comment]
:+1:

**Pull request merged**